kcidb/db/__init__.py

```diff
@@ -134,6 +135,15 @@ def empty(self):
         assert self.is_initialized()
         self.driver.empty()
 
+    def reset(self):
+        """
+        Recreates a database driver instance on request
+        using existing database string
+        Returns:
+            The Recreated driver instance
+        """
+        return Client(self.database)
```

Comment:
This just creates a new `Client()` instance. This method should instead modify the instance it's called for. It should create a new driver instance and store it in `self.driver`. It should not return anything. As I said in my previous comment:

> Essentially you can move the self.driver = ... assignment and the following assertion from __init__() here, and then call this method from __init__() instead.